subprojects/assemble/build-metrics/src/main/kotlin/com/avito/android/plugin/build_metrics/internal/jvm/JvmMetricsSender.kt

```diff
@@ -0,0 +1,49 @@
+package com.avito.android.plugin.build_metrics.internal.jvm
+
+import com.avito.android.build_metrics.BuildMetricTracker
+import com.avito.android.plugin.build_metrics.internal.jvm.LocalVm.GradleDaemon
+import com.avito.android.plugin.build_metrics.internal.jvm.LocalVm.GradleWorker
+import com.avito.android.plugin.build_metrics.internal.jvm.LocalVm.KotlinDaemon
+import com.avito.android.plugin.build_metrics.internal.jvm.LocalVm.Unknown
+import com.avito.android.stats.SeriesName
+import com.avito.android.stats.StatsMetric
+
+internal class JvmMetricsSender(
+    private val tracker: BuildMetricTracker
+) {
+
+    fun send(vm: LocalVm, heapInfo: HeapInfo) {
+        metrics(vm, heapInfo).forEach {
+            tracker.track(it)
+        }
+    }
+
+    private fun metrics(vm: LocalVm, heapInfo: HeapInfo): List<StatsMetric> {
+        val name = SeriesName
+            .create("jvm", "memory")
+            .append(processName(vm))
+
+        return listOf(
+            // Use time metrics instead of gauges to collect statistics for a collection of similar processes.
+            // Namely, Gradle workers and similar.
+            StatsMetric.time(name.append("heap", "used"), heapInfo.heap.usedKb.toLong()),
+            StatsMetric.time(name.append("heap", "committed"), heapInfo.heap.committedKb.toLong()),
+
+            StatsMetric.time(name.append("metaspace", "used"), heapInfo.metaspace.usedKb.toLong()),
+            StatsMetric.time(name.append("metaspace", "committed"), heapInfo.metaspace.committedKb.toLong()),
+        )
+    }
+
+    private fun processName(vm: LocalVm): SeriesName {
+        val name = when (vm) {
+            is GradleDaemon -> "gradle_daemon"
+            is GradleWorker -> "gradle_worker"
+            is KotlinDaemon -> "kotlin_daemon"
+            is Unknown -> vm.name
+                .takeLast(64)
+                .removePrefix(".")
```

Comment:
Could it be `.` symbols not only as prefix?

Reply:
Yes, it can.

Here is a corner case with very long class name or jar name: `very.long.package.ClassName`.
I remove dot to avoid useless `_` in front of series name.
Here may be more smart heuristic to extract class name. 
If we see this metric, we probably should support this process explicitly rather than complex heuristics.